File: Programmers/LEVEL_2/c++/탑.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> heights) {
    vector<int> answer;
    int size = heights.size();

    for (int i = size - 1; i >= 0; i--)
    {
        bool flag = true;
        for (int j = i - 1; j >= 0; j--)
        {
            if (heights[i] < heights[j])
            {
                answer.push_back(j + 1);
                flag = false;
                break;
            }
        }
        if (flag == true)
            answer.push_back(0);
    }
    reverse(answer.begin(), answer.end());

    return answer;
}

File: Programmers/LEVEL_2/탑.js
function solution(heights) {
  var answer = [];

  for (let i = heights.length - 1; i >= 0; i--) {
    let empty = true;

    for (let j = i - 1; j >= 0; j--) {
      if (heights[j] > heights[i]) {
        answer.push(j + 1);
        empty = false;
        break;
      }
    }
    if (empty) answer.push(0);
  }

  answer.reverse();
  return answer;
}

// function solution(heights) {
//     return heights.map((v, i) => {
//         while (i) {
//             i--;
//             if (heights[i] > v) {
//                 return i + 1;
//             }
//         }
//         return 0;
//     });
// }
